addons/sale/models/sale_order.py

```diff
@@ -1349,7 +1349,9 @@ def payment_action_void(self):
 
     def get_portal_last_transaction(self):
         self.ensure_one()
-        return self.transaction_ids._get_last()
+        # In sudo mode to allow the partner of the SO to see transaction
+        # information related to the SO.
+        return self.transaction_ids.sudo()._get_last()
```

Comment:
I'm not a fan of `sudo` being used from the models. I'd rather have the `sudo` in the caller and not in the callee so that another model method doesn't inadvertently receive a sudoed recordset, and we can better delimit the scope of the sudo.

In this case, is it possible to move the sudo call in the calling template(s)?